### ISSUE TITLE
windows-sys does not adhere to the naming conventions for `-sys` crates

### ISSUE THREAD

**Issue by astraw** [kennykerr/blog#1]
`windows-sys` does not adhere to a [naming convention for `-sys` crates](https://doc.rust-lang.org/cargo/reference/build-scripts.html#-sys-packages). I think it would help people if https://kennykerr.ca/rust-getting-started/windows-or-windows-sys.html was updated to explicitly say this. (Relatedly, It is also somewhat surprising to me that `windows` does not depend on `windows-sys`, at least not explicitly.)

**Comment by kennykerr** [comment]
> windows-sys does not adhere to a naming convention for -sys crates. 

I don't follow - what part of the naming convention for `-sys` crates does `windows-sys` not adhere to?

> Relatedly, It is also somewhat surprising to me that windows does not depend on windows-sys, at least not explicitly.

The ABI for the `windows` crate is optimized for its internal consumption. It would be more complicated and take longer to compile if the `windows` crate depended on the `windows-sys` crate.

**Comment by astraw** [comment]
> I don't follow - what part of the naming convention for `-sys` crates does `windows-sys` not adhere to?

The most important aspect of the convention which encouraged me to open this issue is that typically the crate `foo` depends on `foo-sys`. The Cargo book which I linked mentions "It is common to have a companion package without the -sys suffix that provides a safe, high-level abstractions on top of the sys package." but this is not the relationship between `windows` and `windows-sys` as `windows` does not depend on `windows-sys`.

Also, as far as I can tell, there is no `windows.dll` that is linked. (But I didn't look really deeply and and I'm no Windows expert, so maybe I got this aspect wrong.)

> > Relatedly, It is also somewhat surprising to me that windows does not depend on windows-sys, at least not explicitly.
> 
> The ABI for the `windows` crate is optimized for its internal consumption. It would be more complicated and take longer to compile if the `windows` crate depended on the `windows-sys` crate.

I certainly believe there are great reasons for this organization - just the mismatch between my understanding of the naming convention and the actual implementations was confusing to me until I figured out that my understanding of the convention is not being followed here.
